### ISSUE TITLE
EnableW3CDistributedTracing - Version and TraceFlags are inaccessible

### ISSUE THREAD

**Issue by chris-rogala** [Microsoft/ApplicationInsights-aspnetcore#773]
While working on enabling this in a service, we noticed the Version and TraceFlags are not accessible in the RequestTelemetry.  

## Repro Steps
1. Create a ASP .NET Core Web Application
2. Choose the API template (ASP.NET Core 2.1)
3. In Package Manager Console, install Microsoft.ApplicationInsights.AspNetCore
`Install-Package Microsoft.ApplicationInsights.AspNetCore -Version 2.5.0`
4. In Package Manager Console, install Microsoft.ApplicationInsights.DependencyCollector
`Install-Package Microsoft.ApplicationInsights.DependencyCollector -Version 2.8.0`
5. Enable EnableW3CDistributedTracing in Startup.cs

```
public void ConfigureServices(IServiceCollection services)
{
    services.AddApplicationInsightsTelemetry(o => 
        o.RequestCollectionOptions.EnableW3CDistributedTracing = true );
    // ....
}
```
6. In one of the Get method in Controllers/ValuesController.cs, set a local variable to evaluate the RequestTelemetry for the request.

```
//Add using statement
using Microsoft.ApplicationInsights.DataContracts;

//...
var requestTelemetry = Request.HttpContext.Features.Get<RequestTelemetry>();
```

7. Add a breakpoint to the return statement to evaluate requestTelemetry
8. Start application
9. Send the following request to the service (I used PostMan).  Here is the cURL statement:

```
GET /api/values HTTP/1.1
Host: localhost
traceparent: 00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01
tracestate: rojo=00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01,congo=BleGNlZWRzIHRohbCBwbGVhc3VyZS4
cache-control: no-cache
```
10. Look for Version "00" in the requestTelemetry variable
11. Look for TraceFlags "01" in the requestTelemetry variable

##  Actual Behavior
I was able to find the traceid, spanid and the tracestate in the requestTelemetry, but I was not able to find the version and the traceflags.  

```
var traceid = requestTelemetry.Context.Operation.Id;
var spanid = requestTelemetry.Context.Operation.ParentId.Split('.')[1];
var tracestate = requestTelemetry.Context.Properties["w3c_tracestate"];
```

Also, it should be noted that Properties is marked as obsolete:
```
[Obsolete("Use GlobalProperties to set global level properties. For properties at item level, use ISupportProperties.Properties.")]
public IDictionary<string, string> Properties { get; }
```

## Expected Behavior
The version and traceflags should be accessible either in RequestContext (preferably) or another class.  If these are already accessible in another class, please provide direction.  Also, tracestate is accessible in the Context.Properties, but once this is obsolete, it will no longer be accessible.

The reason access is needed for the flags and the version, is so the traceparent and tracestate can be added to outbound requests.

## Version Info 
SDK Version : Microsoft.ApplicationInsights.AspNetCore -Version 2.5.0
Microsoft.ApplicationInsights.DependencyCollector -Version 2.8.0
.NET Version :	ASP.NET Core 2.1 						
How Application was onboarded with SDK(VisualStudio/StatusMonitor/Azure Extension) : VisualStudio
OS : Windows 10 (64-bit)
Hosting Info (IIS/Azure WebApps/ etc) : IISExpress (local) && Azure WebApps (deployed)

**Comment by lmolkova** [comment]
ApplicaitonInsights will do outbound header propagation when W3C is enabled - you don't have to implement it on your own. Please let me know if something does not work for you and I'll fix it.

Support for W3C is experimental in AppicaitonInsights. We are working with .NET team to make Activity implement W3C tracing and trace flags will become available. Regarding version, AI implements 00 and there is no point in exposing version - it will likely be const anyway.

You may still access trace flags from the Activity.Current.Tags, in w3c_sampled (all tag names - https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/Common/W3C/W3CConstants.cs).

RequestTelemtery.Context.Properties are obsolete, but they are the same as RequestTelemtery.Properties which are not going anywhere.

I'm closing this issue for now, but feel free to reopen. And please let me know if you need more info, I'd like to understand your scenario and the need to re-inject headers better.

**Comment by chris-rogala** [comment]
Per the document [https://w3c.github.io/distributed-tracing/report-trace-context.html#overview](https://w3c.github.io/distributed-tracing/report-trace-context.html#overview), it states at the end the following:

> Libraries and platforms MUST propagate traceparent and tracestate headers to guarantee that trace will not be broken.

While the package does handle outbound header propagation to Application Insights, it does not appear to add them to outbound HttpRequestMessages in the HttpClient.SendAsync.  I did a local test where I had one application send an Http request to another application.  I did not see the traceparent or tracestate in the headers of the request.  That said, I did see the Request-Id being sent but that;s it.

Additionally, with respect to asynchronous operations, like sending a message to a service bus queue, I would think the traceparent and tracestate should be propagated to those types of outbound requests.

This is why I believe being able to access the version would be beneficial.


Currently, it cannot be assumed that the version will always be 00 since it is part of the traceparent header and could be generated from another system,  The document only states: 

> Version (version) is a 1 byte representing an 8-bit unsigned integer. Version 255 is invalid. Current specification assumes the version is set to 00.

In the Base mutations section [https://w3c.github.io/distributed-tracing/report-trace-context.html#mutating-the-traceparent-field](https://w3c.github.io/distributed-tracing/report-trace-context.html#mutating-the-traceparent-field), it does not reference the version.  I would assume the version provided should be propagated to other services.
